app/controllers/katello/api/v2/sync_plans_controller.rb

```diff
@@ -67,12 +67,26 @@ def create
     param_group :sync_plan
     def update
       sync_date = sync_plan_params.try(:[], :sync_date).try(:to_time)
-
       if !sync_date.nil? && !sync_date.is_a?(Time)
         fail _("Date format is incorrect.")
       end
+      rec_logic_change = (sync_plan_params["sync_date"] && @sync_plan.sync_date != sync_plan_params["sync_date"]) || (sync_plan_params["interval"] && @sync_plan.interval != sync_plan_params["interval"])
+      toggle_enable = sync_plan_params["enabled"] && @sync_plan.enabled != sync_plan_params["enabled"]
+
+      @sync_plan.update_attributes(sync_plan_params) if sync_plan_params
+
+      if rec_logic_change
+        old_recurring_logic = @sync_plan.foreman_tasks_recurring_logic
+        sync_date ||= @sync_plan.sync_date
+        interval = sync_plan_params[:interval] || @sync_plan.interval
+        recurring_logic = add_recurring_logic_to_sync_plan(sync_date, interval)
+        @sync_plan.foreman_tasks_recurring_logic_id = recurring_logic.id
+        recurring_logic.start_after(::Actions::Katello::SyncPlan::Run, sync_date.to_time, @sync_plan)
+        old_recurring_logic.cancel
+      end
+      @sync_plan.save!
+      @sync_plan.recurring_logic.enabled = @sync_plan.enabled if toggle_enable
```

Comment:
I think we want to get as much of this logic out of the controller and into the model as possible, possibly via after_save hooks?